libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -563,6 +563,31 @@ impl Greenlight {
 
 #[tonic::async_trait]
 impl NodeAPI for Greenlight {
+    fn node_credentials(&self) -> NodeResult<Option<NodeCredentials>> {
+        let encryption_key = Self::derive_bip32_key(
+            self.sdk_config.network,
+            &self.signer,
+            vec![ChildNumber::from_hardened_idx(140)?, ChildNumber::from(0)],
+        )?
+        .to_priv()
+        .to_bytes();
+        let encryption_key_slice = encryption_key.as_slice();
+
+        match self.persister.get_gl_credentials()? {
+            Some(creds) => match sym_decrypt(encryption_key_slice, creds.as_slice()) {
+                Some(creds) => {
+                    let credentials: GreenlightCredentials =
+                        serde_json::from_slice(creds.as_slice()).map_err(|e| {
+                            NodeError::Generic(anyhow!("Unable to parse credentials: {e}"))
+                        })?;
+                    Ok(Some(NodeCredentials::Greenlight { credentials }))
+                }
+                None => Ok(None),
```

Comment:
Failing to decrypt should throw an error, see https://github.com/breez/breez-sdk/blob/6b74de2ae9c939789fb7eb5301814cd961e2ac6b/libs/sdk-core/src/greenlight/node_api.rs#L117C56-L117C56